lib/client.js

```diff
@@ -254,6 +254,15 @@ class AssetComputeClient extends EventEmitter {
         });
     }
 
+    async register() {
+        // Register I/O event type and journal, update journal in eventEmitter
+        const { journal } = await assetCompute.register();
+        this.eventEmitter.journal = journal;
```

Comment:
Hmm, how would one use it? Looking at https://github.com/adobe/asset-compute-client/blob/34df3483f3f2b0404b27bca8072358ca5a0654f1/lib/client.js#L311-L320 which calls `register()` and only _afterwards_ creates an instance of this class.

It seems the design of this class is slightly confusing. One doesn't use it's constructor to create it (as very internal implementation objects have to be passed in), but instead `createAssetComputeClient()`. I feel this could be cleaned up.

Reply:
The reason I added these two methods were so a user can directly call `.register()` and `.unregister()` on the `AssetComputeClient` instance.

I did not want to change the existing functionality in `createAssetComputeClient()`. I thought the behavior would go like this:

```js
const assetComputeClient = await createAssetComputeClient(integration, options);
// no need to call register because client takes care of that
// process renditions
await assetComputeClient.process(...);

// unregister journal 
await assetComputeClient.unregister();
...
// re-register journal
await assetComputeClient.register(); // if the new journal did not get set in the eventEmitter, the next call to process would not work
await assetComputeClient.process(...);
```

If those two methods were not in the `AssetComputeClient`, you would have to call them like this, which feels unnatural:
```js
await assetComputeClient.assetCompute.unregister();
await assetComputeClient.assetCompute.register(); 
```

What would you recommend I do to clean this up?

Reply:
I see.

I feel like the unregister/register would create a new EventEmitter as the journal changes, and this all happens internally of the class, without having the outside like a `createAssetComputeClient()` helper know about it. As it stands, a client can do `new AssetComputeClient(....)` and set those internal values as constructor arguments, which is weird.

I would just make the constructor do what `createAssetComputeClient` does, something like that:

```js
const assetComputeClient = new AssetComputeClient(integration, options);
// process renditions
await assetComputeClient.process(...);

// unregister journal 
await assetComputeClient.unregister();

// ... all the same
```

Then drop the `createAssetComputeClient` function. Still do an implicit `register()` in the class, if not in the constructor (because it cannot be async), do it lazy with sth like `await ensureRegistered()` at the start of each API method (probably only check for a first time call after the constructor, so that it wouldn't be triggered after an `unregister()` again.

Reply:
Ok, that makes sense.

The only issue I see with it is creating the access token (which is done in createAssetComputeClient), is async: https://github.com/adobe/asset-compute-client/pull/8/files#diff-50cfa59973c04321b5da0c6da0fdf4feL298. 

"Still do an implicit register() in the class, if not in the constructor (because it cannot be async),"
Also are you saying to put the `register()` call in the constructor but not with an await?:
```
constructed(){
...
this.assetCompute.register()
}
```

Reply:
IMO this implementation is not really cleaner than the `createAssetComputeClient` from before, but take a look and let me know. 

I made a separate `intialize()` method that takes care of creating the access token and setting up the `AssetCompute` class instance. This gets called implicitly inside the `register()` method. I made a boolean variable inside the class `registered` to determine easily by the other api calls if register has been called.

I adjusted and kept the old `createAssetComputeClient` for compatibility reasons since this is now OS.

Take a look and let me know what you think